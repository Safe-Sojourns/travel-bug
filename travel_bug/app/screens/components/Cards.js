import React from 'react';
import {StyleSheet, Text, View, ImageBackground} from 'react-native';
// import CollapsibleCard from './CollapsibleCard';
import CollapsibleView from '@eliav2/react-native-collapsible-view';

export default function CardDetails() {
  return (
    <View>
      <CollapsibleView title={<Card />} style={styles.app}>
        <ImageBackground
          source={require('../assets/rome.jpg')}
          style={{width: 320, height: 200}}
        />
        <Text />
        <Text>
          Come see the beautiful architecture. Our tour begins at the Trevi
          Fountain!
        </Text>
        <Text />
        <Text>Location</Text>
        <Text>Cost: </Text>
        <Text>Transport: </Text>
      </CollapsibleView>
    </View>
  );
}

const Card = () => {
  return (
    <View style={styles.container}>
      <Text testID="activity" style={styles.title}>
        Flight to Rome
      </Text>
      <Text style={styles.inline}>
        <Text testID="time" style={styles.time}>
          9.00am
        </Text>
        <Text>{'        '}</Text>
        <Text>{'        '}</Text>
        <Text>{'        '}</Text>
        <Text>{'     '}</Text>
        <Text>{'     '}</Text>
        <Text>{'     '}</Text>
        <Text style={styles.optional}>[OPTIONAL]</Text>
      </Text>
    </View>
  );
};

const styles = StyleSheet.create({
  app: {
    borderRadius: 10,
    width: 320,
    display: 'flex',
    left: 20,
  },
  paragraph: {
    fontSize: 14,
    lineHeight: 19,
  },
  container: {
    alignItems: 'stretch',
    width: 300,
    display: 'flex',
  },
  inline: {
    padding: 5,
  },
  title: {
    display: 'flex',
    alignItems: 'flex-start',
    fontSize: 20,
    fontWeight: 'bold',
  },
  time: {
    fontWeight: 'bold',
    color: 'red',
    margin: 20,
  },
  optional: {
    fontSize: 12,
    color: 'black',
  },
});
