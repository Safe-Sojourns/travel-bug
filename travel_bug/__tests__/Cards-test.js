import React from 'react';
import renderer from 'react-test-renderer';
import Cards from '../app/screens/components/Cards.js';

jest.useFakeTimers();

const tree = renderer.create(<Cards />);

test('Cards renders correctly', () => {
  expect(tree).toMatchSnapshot();
});

test('Time exists', () => {
  const time = tree.root.findByProps({testID: 'time'}).props;
  expect(time).toBeDefined();
});

test('Activity exists', () => {
  const activity = tree.root.findByProps({testID: 'activity'}).props;
  expect(activity).toBeDefined();
});
