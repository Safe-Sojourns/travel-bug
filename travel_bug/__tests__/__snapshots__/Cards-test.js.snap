// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Cards renders correctly 1`] = `
<View>
  <View
    accessible={true}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "alignItems": "center",
        "borderColor": "grey",
        "borderRadius": 10,
        "borderStyle": "solid",
        "borderWidth": 1,
        "display": "flex",
        "marginHorizontal": 10,
        "marginVertical": 5,
        "opacity": 1,
        "padding": 5,
        "width": 320,
      }
    }
  >
    <View
      style={
        Object {
          "alignItems": "center",
          "flexDirection": "row",
        }
      }
    >
      <View
        style={
          Object {
            "transform": Array [
              Object {
                "rotate": "0deg",
              },
            ],
          }
        }
      >
        <RNSVGSvgView
          bbHeight={24}
          bbWidth={24}
          focusable={false}
          height={24}
          style={
            Array [
              Object {
                "backgroundColor": "transparent",
                "borderWidth": 0,
              },
              Object {
                "flex": 0,
                "height": 24,
                "width": 24,
              },
            ]
          }
          width={24}
        >
          <RNSVGGroup>
            <RNSVGPath
              d="M4 8 12 16 20 8"
              fill={null}
              propList={
                Array [
                  "fill",
                  "stroke",
                  "strokeWidth",
                ]
              }
              stroke={4278190080}
              strokeWidth="1.5"
            />
          </RNSVGGroup>
        </RNSVGSvgView>
      </View>
      <View
        style={
          Object {
            "alignItems": "stretch",
            "display": "flex",
            "width": 300,
          }
        }
      >
        <Text
          style={
            Object {
              "alignItems": "flex-start",
              "display": "flex",
              "fontSize": 20,
              "fontWeight": "bold",
            }
          }
          testID="activity"
        >
          Flight to Rome
        </Text>
        <Text
          style={
            Object {
              "padding": 5,
            }
          }
        >
          <Text
            style={
              Object {
                "color": "red",
                "fontWeight": "bold",
                "margin": 20,
              }
            }
            testID="time"
          >
            9.00am
          </Text>
          <Text>
                    
          </Text>
          <Text>
                    
          </Text>
          <Text>
                    
          </Text>
          <Text>
                 
          </Text>
          <Text>
                 
          </Text>
          <Text>
                 
          </Text>
          <Text
            style={
              Object {
                "color": "black",
                "fontSize": 12,
              }
            }
          >
            [OPTIONAL]
          </Text>
        </Text>
      </View>
    </View>
  </View>
</View>
`;
